src/main/kotlin/no/nav/syfo/person/skjermingskode/SkjermingskodeService.kt

```diff
@@ -1,32 +1,56 @@
 package no.nav.syfo.person.skjermingskode
 
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
 import no.nav.syfo.client.pdl.PdlClient
 import no.nav.syfo.client.skjermedepersonerpip.SkjermedePersonerPipClient
 import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.person.api.domain.PersonInfo
 import no.nav.syfo.person.api.domain.Skjermingskode
 
 class SkjermingskodeService(
     private val skjermedePersonerPipClient: SkjermedePersonerPipClient,
     private val pdlClient: PdlClient,
 ) {
-    suspend fun hentBrukersSkjermingskode(
+    suspend fun hentSkjermingskodeForPersonidenter(
         callId: String,
-        personIdent: PersonIdentNumber,
+        personidenter: List<PersonIdentNumber>,
         token: String,
-    ): Skjermingskode? {
-        val hasAdressebeskyttelse = pdlClient.hasAdressebeskyttelse(callId = callId, personIdent = personIdent)
+    ) =
+        personidenter.map { personident ->
+            val skjermingskode = hentBrukersSkjermingskode(
+                callId = callId,
+                personident = personident,
+                token = token,
+            )
+            Pair(personident, skjermingskode)
+        }.map { (personident, skjermingskode) ->
+            PersonInfo(
+                fnr = personident.value,
+                skjermingskode = skjermingskode.await(),
+            )
+        }
```

Comment:
`map`'er i to omganger her for å sende avgårde alle kallene her før man fanger de igjen når man lager `PersonInfo` objektet.

Reply:
Smart!